query-engine/connectors/sql-query-connector/src/filter/visitor.rs

```diff
@@ -1324,7 +1348,10 @@ impl JsonFilterExt for (Expression<'static>, Expression<'static>) {
         match (value, target_type) {
             // string_starts_with (value)
             (ConditionValue::Value(value), JsonTargetType::String) => {
-                let starts_with = expr_string.like(format!("{value}%"));
+                let starts_with = match query_mode {
+                    QueryMode::Default => expr_string.like(format!("{value}%")),
+                    QueryMode::Insensitive => expr_string.compare_raw("ILIKE", format!("{value}%")),
```

Comment:
Can we choose another name for the currently called "query mode"? How about `case: 'sensitive' | 'insensitive'`, where `case: 'sensitive'` is the default option?

Reply:
After adding `case` enum, it feels a bit incosistent with the rest of `where` api. Especially when looking into cases like:

```ts
prisma.recipe.findMany({
  where: {
    allergens: {
      not: { contains: "peanuts" },
      mode: "insensitive", // <-- mode
    },
    title: {
      path: [locale],
      string_contains: filter.search,
      case: "insensitive", // <-- case
    },
  },
})
```

Reply:
Hi @lubosmato, after talking with a colleague, and observing this output, I stand corrected. I didn't realise `QueryMode` was an already existing `enum` in out codebase (https://github.com/prisma/prisma-engines/blob/119ee813207221af5e41c432b375375ad4058474/query-engine/query-structure/src/filter/scalar/mod.rs#L29). So, let's stick with `mode: 'default' | 'insensitive'`:)

Reply:
Hi @jkomyno, thank you for reply. I reverted the commit with `case` enum.